clients/admin-ui/src/features/datastore-connections/system_portal_config/forms/fields/DatasetConfigField/YamlEditor.tsx

```diff
@@ -64,7 +66,12 @@ const YamlEditor = ({
       setIsTouched(true);
       validate(value as string);
       setIsEmptyState(!value || value.trim() === "");
-      onChange(yaml.load(value || "", { json: true }) as Dataset);
+      const yamlDoc = yaml.load(value || "", { json: true });
+      if (Array.isArray(yamlDoc)) {
+        onChange(yamlDoc[0] as Dataset);
+      } else {
+        onChange(yamlDoc as Dataset);
+      }
```

Comment:
The YAML can be parsed into a single dataset (as when creating a new dataset in the system integrations tab) or into an array with one element (as when editing datasets from the manage datasets view).  This is just to make TypeScript happy.